packages/flutter/lib/src/material/dialog.dart

```diff
@@ -216,6 +217,16 @@ class AlertDialog extends StatelessWidget {
   /// from the [actions].
   final List<Widget> actions;
 
+  /// An (optional) name for the dialog used by accessibility frameworks.
+  ///
+  /// On android this would usually be a short summary of the contents. For
+  /// example - "App permissions" for a dialog containing permission
+  /// information.
+  ///
+  /// On iOS it is not common to provide a semantic name, since an announcement
+  /// chime is expected instead.
```

Comment:
Maybe add that there are exception to this and list an example, e.g. Alerts?

Reply:
Rephrased this so that it is about dialogs and not the semantics in general